job/server/src/main/java/alluxio/job/plan/replicate/SetReplicaDefinition.java

```diff
@@ -168,8 +170,17 @@ private void replicate(SetReplicaConfig config, RunTaskContext context) throws E
     // to avoid the the race between "replicate" and "rename", so that even a file to replicate is
     // renamed, the job is still working on the correct file.
     URIStatus status = context.getFileSystem().getStatus(new AlluxioURI(config.getPath()));
-
-    JobUtils.loadBlock(status, context.getFsContext(), config.getBlockId(), null, false);
+    try {
+      JobUtils.loadBlock(status, context.getFsContext(), config.getBlockId(), null, false);
+    } catch (IOException e) {
+      LOG.warn("Replication of {} failed, reduce min replication to 1 and unpin.",
+          status.getPath());
+      SetAttributePOptions.Builder optionsBuilder =
+          SetAttributePOptions.newBuilder();
+      context.getFileSystem().setAttribute(new AlluxioURI(config.getPath()),
```

Comment:
will this throw exception? how do we handle it?

Reply:
addressed as a supressed exception